### PR TITLE
Added typescript-mode to ts[x] autolist. & init-scala-metals.el based on Scala metals. Ref: https://scalameta.org/metals/

### PR DESCRIPTION
**Pull request by lanyusan**
Added init-scala-metals.el based on Scala metals. Ref: https://scalameta.org/metals/

### CODE REVIEW

File: lisp/init-scala-metals.el [hunk 1]
```diff
@@ -0,0 +1,43 @@
+;; init-web.el --- Initialize web configurations.	-*- lexical-binding: t -*-
```
Comment (seagle0128) [inline lisp/init-scala-metals.el]: The docs should be updated.

File: lisp/init-scala-metals.el [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  (require 'init-custom))
+
+(use-package scala-mode
+  :mode (("\\.s\\(cala\\|bt\\)$"    . scala-mode))
+  :hook ((scala-mode-hook lsp)))
+
+(provide 'init-scala-metals)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; init-web.el ends here
```
Comment (seagle0128) [inline lisp/init-scala-metals.el]: The docs should be updated.

File: init.el [hunk 3]
```diff
@@ -170,6 +170,7 @@ decrease this. If you experience stuttering, increase this.")
 (require 'init-dart)
 (require 'init-elixir)
 (require 'init-web)
+(require 'init-scala-metals)
```
Comment (seagle0128) [inline init.el]: I think `init-scala` is enough.

File: lisp/init-prog.el [hunk 4]
```diff
@@ -36,26 +36,28 @@
...
-                                         (">=" . ?≥)
-                                         ("=<<" . (?= (Br . Bl) ?≪))
-                                         (">>=" . (?≫ (Br . Bl) ?=))
-                                         ("<=<" . ?↢)
-                                         (">=>" . ?↣)
-                                         ("&&" . ?∧)
-                                         ("||" . ?∨)
-                                         ("not" . ?¬)))
-  (setq prettify-symbols-unprettify-at-point 'right-edge))
+  ;; :init
```
[the remark this answers is not in the document]
Comment (lanyusan) [reply lisp/init-prog.el]: A mistake. I don't like beautify those symbols so commented them out. It wasn't meant for merging.

File: lisp/init-scala-metals.el (hunk 1 above)
Comment (lanyusan) [reply lisp/init-scala-metals.el]: Ok. I will push a new version for merge.

**Comment by seagle0128** [comment]
I added a new option `centaur-prettify-symbols`.

Why was this closed?

File: lisp/init-scala.el [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+;;
+
+;;; Code:
+
+(eval-when-compile
+  (require 'init-custom))
+
+(use-package scala-mode
+  :mode (("\\.s\\(cala\\|bt\\)$"    . scala-mode))
+  :hook ((scala-mode-hook lsp)))
```
Comment (seagle0128) [inline lisp/init-scala.el]: This hook is invalid.

Should be `(scala-mode . lsp)`.

**Pull request closed without merging**